sonar-docs/analysis/scan/sonarscanner-for-msbuild.md

```diff
@@ -252,7 +252,9 @@ Some analysis parameters can be set for a single MSBuild project by adding them
 
 **Analyzing languages other than C# and VB**
 
-By default, SonarScanner for .NET will only analyze C# and VB files in your project. To enable the analysis of other types of files, these files must be listed in the MSBuild project file (the `.csproj` or `.vbproj` file).
+For .NET Core, .NET 5.0 and later projects that use Microsoft.NET.Sdk, SonarScanner for .NET will analyze any files that are not excluded and that can be analyzed by the available language plugins. 
```

Comment:
I think this still needs some work:

The reference to `Microsoft.NET.Sdk` is confusing: the [Microsoft docs](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/overview) list multiple available SDKs.

The reference to `.NET Core` and `.NET 5.0` isn't accurate - it's valid to have a new-format MSBuild project that targets the .NET Framework.

The important thing is thing is whether the project is using the old-style or the new-style ("SDK-style") MSBuild project format.

If the project is using the old-style format then all files have to explicitly referenced in the project.

If the project is using the new-style format then by default certain file types will be implicitly included (unless the user has explictly opted out of the default behaviour).

Reply:
Thanks for the clarification. I've updated the PR and tried to keep it simple without losing any accuracy. I've kept a mention of[ .NET Core and 5 as some devs may not be aware of when the SDK-style change was introduced](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/overview).